Memory leak using openssl-3 filed as community bug 3910. 
https://bugs.ntp.org/show_bug.cgi?id=3910

This patch may be removed when upgrading to a version of NTP with this bug
fixed in it.

--- libntp/a_md5encrypt.c
+++ libntp/a_md5encrypt.c
@@ -169,7 +169,11 @@ make_mac(
 
 		if (ctx)
 			EVP_MD_CTX_free(ctx);
-	}
+		if (octx)
+			OSSL_LIB_CTX_free(octx);
+		if (type)
+			EVP_MD_free(type);
+	}
 
 #else /* !OPENSSL follows */
 
